pinakes/main/approval/services/update_request.py

```diff
@@ -218,15 +218,31 @@ def _persist_request(self, time_field=None):
     def _should_auto_approve(self):
         if self.request.is_parent():
             return False
+        if self._group_approved():
+            return True
         return not bool(self.request.workflow)
 
     def _should_auto_notify(self):
         if self.request.is_parent():
             return False
+        if self._group_approved():
+            return True
         return not self._external_processable()
 
     def _external_processable(self):
         return bool(
             self.request.workflow
             and self.request.workflow.template.process_method
         )
+
+    def _group_approved(self):
+        """if the current group already approved another leaf request"""
+        if not self.request.group_ref or self.request.is_parent():
```

Comment:
Do you need this check? the final `return False` has cover this case, right?

Reply:
it returns earlier. we need to check `group_ref` otherwise it may falsely return true in the spec tests.